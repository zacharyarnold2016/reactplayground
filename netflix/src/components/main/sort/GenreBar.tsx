import React from "react";
import GenreOption from "./GenreOption";
import { GENRES } from "../../../interfaces/enums/lists.enums";

const GenreBar = () => {
  const genreOptions = Object.values(GENRES);
  return (
    <div className="genreBar">
      {genreOptions.map((genre) => {
        return (
          <li className="sortOption" key={genre.toString()}>
            <GenreOption genre={genre} />
          </li>
        );
      })}
    </div>
  );
};

export default GenreBar;
