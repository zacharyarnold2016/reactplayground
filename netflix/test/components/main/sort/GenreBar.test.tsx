// @ts-nocheck
import { expect } from "@jest/globals";
import renderer from "react-test-renderer";
import GenreBar from "../../../../src/components/main/sort/GenreBar";
import { useNavigate, useParams } from "react-router";
import * as route from "react-router-dom";

jest.mock("react-router");
jest.mock("react-router-dom");
afterAll(() => {
  jest.unmock("react-router");
  jest.unmock("react-router-dom");
});

const spyFunction = jest.fn();

describe("GenreBar", () => {
  test("SnapShot", () => {
    route.useSearchParams.mockReturnValue([{ get: spyFunction }, spyFunction]);
    const component = renderer.create(<GenreBar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
