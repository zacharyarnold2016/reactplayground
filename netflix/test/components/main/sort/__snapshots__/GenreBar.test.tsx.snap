// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreBar SnapShot 1`] = `
<div
  className="genreBar"
>
  <li
    className="sortOption"
  >
    <h4
      onClick={[Function]}
    >
      All
    </h4>
  </li>
  <li
    className="sortOption"
  >
    <h4
      onClick={[Function]}
    >
      Action
    </h4>
  </li>
  <li
    className="sortOption"
  >
    <h4
      onClick={[Function]}
    >
      Drama
    </h4>
  </li>
  <li
    className="sortOption"
  >
    <h4
      onClick={[Function]}
    >
      Comedy
    </h4>
  </li>
  <li
    className="sortOption"
  >
    <h4
      onClick={[Function]}
    >
      Horror
    </h4>
  </li>
  <li
    className="sortOption"
  >
    <h4
      onClick={[Function]}
    >
      Documentary
    </h4>
  </li>
</div>
`;
